data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceCustomConsumer.java

```diff
@@ -176,12 +170,27 @@ public <T> void consumeRecords() throws Exception {
                 }
             }
         } catch (AuthenticationException e) {
-            LOG.warn("Authentication Error while doing poll(). Will retry after 10 seconds", e);
+            LOG.warn("Access Denied while doing poll(). Will retry after 10 seconds", e);
             Thread.sleep(10000);
+        } catch (RecordDeserializationException e) {
+            LOG.warn("Serialization error - topic {} partition {} offset {}, seeking past the error record",
+                     e.topicPartition().topic(), e.topicPartition().partition(), e.offset());
+            consumer.seek(e.topicPartition(), e.offset()+1);
```

Comment:
is it really required to explicitly seek past this record? can we not just log exception, count a metric and commit offset as usual?